CRM/Eck/BAO/EckEntityType.php

```diff
@@ -14,8 +14,23 @@
 +--------------------------------------------------------*/
 
 use CRM_Eck_ExtensionUtil as E;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Civi\Core\Event\PreEvent;
+use Civi\Core\Event\PostEvent;
+use Civi\Api4\CustomGroup;
+use Civi\Api4\EckEntity;
+use Civi\Api4\OptionValue;
+use Civi\Api4\Managed;
 
-class CRM_Eck_BAO_EckEntityType extends CRM_Eck_DAO_EckEntityType {
+class CRM_Eck_BAO_EckEntityType extends CRM_Eck_DAO_EckEntityType implements EventSubscriberInterface {
```

Comment:
implementing the `EventSubscriberInterface` isn't enough to make Civi autoload the class.
There are a few ways to solve that. I think the simplest thing would be to implement `HookInterface` instead of `EventSubscriberInterface` and switch your hooks to the classic pattern using naming conventions instead of `getSubscribedEvents`